spec/services/workflow_errors_reporter_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'rails_helper'
+
+RSpec.describe WorkflowErrorsReporter do
+  let(:put) { 'https://sul-lyberservices-test.stanford.edu/workflow/dor/objects/druid:jj925bx9565/workflows/preservationWF/moab-valid' }
+  let(:headers) { { 'Content-Type' => 'application/xml' } }
+
+  describe '.update_workflow' do
+    let(:error_msg) do
+      { 13 => "PreservedObjectHandler(jj925bx9565, 9, 6570668, <Endpoint:...>) Invalid moab, validation error...ential version directories. Current directories: [\\\"v0001\\\", \\\"v0002\\\", \\\"v0009\\\"]\"]" }
+    end
+
+    let(:body) { "<process name='moab-valid' status='error' errorMessage='#{error_msg}'/>" }
+
+    before do
+      Settings.workflow_services_url = 'https://sul-lyberservices-test.stanford.edu/workflow/'
+      stub_request(:put, put)
+        .with(body: body,
+              headers: headers)
+        .to_return(status: 204, body: "", headers: {})
+    end
+    it 'updates preservationWF with invalid moab errors' do
+      expect(described_class.update_workflow('jj925bx9565', 'moab-valid', error_msg)).to be_an_instance_of(Faraday::Response)
+    end
+
+    it 'rescues for Faraday::Error' do
+      allow(described_class).to receive(:update_workflow).and_raise(Faraday::Error)
+      expect { described_class.update_workflow('jj925bx9565', 'moab-valid', 'Foo() error') }.to raise_error(Faraday::Error)
```

Comment:
I don't understand this test.  The expect statement says you expect to raise a `Faraday::Error` ... so where does the rescue get tested?

Reply:
if i change the test to : `      expect { described_class.update_workflow('jj925bx9565', 'moab-val', 'Foo() error') }.to raise_error(Faraday::Error)
` i think this satisfies the test?

Reply:
(note its `moab-val` instead of `moab-valid`)